docs/app/HOC/pure.js

```diff
@@ -0,0 +1,15 @@
+import React, { Component } from 'react'
+
+import { shallowEqual } from 'src/lib'
+
+const pure = ChildComponent => class extends Component {
+  shouldComponentUpdate(nextProps) {
+    return !shallowEqual(this.props, nextProps)
+  }
+
+  render() {
+    return <ChildComponent {...this.props} />
+  }
+}
+
+export default pure
```

Comment:
I believe this can be replaced by extending [`React.PureComponent`](https://facebook.github.io/react/docs/react-api.html#react.purecomponent), correct?